src/GafferSceneUI/LightTool.cpp

```diff
@@ -158,6 +168,50 @@ void setValueOrAddKey( FloatPlug *plug, float time, float value )
 	}
 }
 
+bool inspectionEditable( const GafferSceneUI::Private::Inspector::Result *inspection )
+{
+	if( !inspection->editable() )
+	{
+		return false;
+	}
+
+	EditScope *editScope = inspection->editScope();
+	if(
+		( editScope && inspection->source() && editScope->isAncestorOf( inspection->source() ) ) ||
+		!editScope
+	)
+	{
+		// If we're editing an EditScope and the source is in the edit scope,
+		// check to see if it can be set. If the source is outside the edit scope,
+		// we'll be creating a new edit in the scope so `inspection->editable()` will
+		// cover that case.
+		// If we're not editing an EditScope, no magic can happen and the editability
+		// comes directly from the source.
+		Plug *p = nullptr;
+		if( auto nameValuePlug = runTimeCast<NameValuePlug>(inspection->source() ) )
+		{
+			p = nameValuePlug->valuePlug();
+		}
+		else if( auto tweakPlug = runTimeCast<TweakPlug>( inspection->source() ) )
+		{
+			p = tweakPlug->valuePlug();
+		}
+		else
+		{
+			p = inspection->source();
+		}
```

Comment:
The `activeValuePlug()` function edits `TweakPlug.enabledPlug` and `NameValuePlug.enabledPlug` as well, so checking only the `valuePlug()` is too limited. When we move this to the Inspector, perhaps it's better to just consider all the children of `source()` since we don't really know what anyone will do with the result of `acquireEdit()`?